common/ferrostar/src/routing_adapters/osrm/mod.rs

```diff
@@ -48,7 +48,13 @@ impl RouteResponseParser for OsrmResponseParser {
                 .map(|route| Route::from_osrm(route, &res.waypoints, self.polyline_precision))
                 .collect::<Result<Vec<_>, _>>()
         } else {
-            Err(ParsingError::InvalidStatusCode { code: res.code })
+            let error_description = match res.message {
+                Some(message) => format!("{}: {}", res.code, message),
+                None => res.code,
+            };
+            Err(ParsingError::InvalidStatusCode {
+                code: error_description,
```

Comment:
I'd suggest keeping the code and _adding_ a description field. Also, the struct field doesn't match what it stores now though (it says `code` but it's really a combined description) :)